wallet/src/de/schildbach/wallet/service/InactivityNotificationService.java

```diff
@@ -46,7 +47,11 @@
  */
 public final class InactivityNotificationService extends IntentService {
     public static void startMaybeShowNotification(final Context context) {
-        context.startService(new Intent(context, InactivityNotificationService.class));
+        final Intent intent = new Intent(context, InactivityNotificationService.class);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
```

Comment:
This `if` block can be replaced with ContextCompat.startForegroundService(context, intent). See [this](https://medium.com/google-developers/migrating-mediastyle-notifications-to-support-android-o-29c7edeca9b7).